causalml/inference/tree/uplift.pyx

```diff
@@ -297,22 +314,44 @@ class UpliftTreeClassifier:
+                                 self.evaluationFunction == self.evaluate_IT or self.evaluationFunction == self.evaluate_CIT):
+            raise ValueError("The DDP, IDDP, IT, and CIT approach can only cope with two class problems, that is two different treatment "
                              "options (e.g., control vs treatment). Please select another approach or only use a "
                              "dataset which employs two treatment options.")
 
+        if self.honesty:
+            try:
+                X, X_est, treatment_idx, treatment_idx_est, y, y_est = sklearn.model_selection.train_test_split(X,
+                                                                                                                treatment_idx,
+                                                                                                                y,
+                                                                                                                stratify=[treatment_idx, y],
+                                                                                                                test_size=self.estimation_sample_size,
+                                                                                                                shuffle=True,
+                                                                                                                random_state=self.random_state)
+            except ValueError:
+                X, X_est, treatment_idx, treatment_idx_est, y, y_est = sklearn.model_selection.train_test_split(X,
+                                                                                                                treatment_idx,
+                                                                                                                y,
+                                                                                                                test_size=self.estimation_sample_size,
+                                                                                                                shuffle=True,
+                                                                                                                random_state=self.random_state)
+
         self.fitted_uplift_tree = self.growDecisionTreeFrom(
             X, treatment_idx, y,
             max_depth=self.max_depth, min_samples_leaf=self.min_samples_leaf,
             depth=1, min_samples_treatment=self.min_samples_treatment,
             n_reg=self.n_reg, parentNodeSummary=None
         )
 
-        self.feature_importances_ = np.zeros(X.shape[1])
-        for col, imp in self.feature_imp_dict.items():
-            self.feature_importances_[col] = imp
-        self.feature_importances_ /= self.feature_importances_.sum()  # normalize to add to 1
+        if self.honesty:
+            self.honestApproach(X_est, treatment_idx_est, y_est)
+
+        with np.errstate(divide='ignore',invalid='ignore'):
```

Comment:
Similarly, if there's an error, e.g., all feature importances being zero in this case, we'd like the code to fail with a proper error message.